buildconfig/stubs/pygame/draw.pyi

```diff
@@ -33,6 +33,10 @@ def circle(
     draw_bottom_left: bool = False,
     draw_bottom_right: bool = False,
 ) -> Rect: ...
+def circles(
+        surface: Surface,
+        draw_sequence: Sequence[Tuple[ColorValue, Coordinate, int, Optional[int]]], /
```

Comment:
So one slightly annoying thing about this `Optional[int]` type hint is that it means "An integer value or `None`, is expected here". If you put only three parameters into the tuple you get a typing error, but it will still work.

Perhaps this hint should be:

`Sequence[Union[Tuple[ColorValue, Coordinate, int, int], Tuple[ColorValue, Coordinate, int]]]`

Reply:
About the perf tests, i think we have the same issue here that we have with fblits or blits. Looping will be more efficient with smaller circles or hollow ones so performance gains can vary significantly.

Reply:
btw i still have to add many tests to this PR and all my other PRs in #3269. Might discover more issues when i do so. Just waiting for reviews like this to make sure i don't spend time where it's not wanted/needed.